framework/util/page_guard_manager.cpp

```diff
@@ -230,15 +233,102 @@ PageGuardManager::~PageGuardManager()
     }
 }
 
+#if !defined(WIN32)
+constexpr uint32_t PageGuardManager::signal_handler_watcher_max_restores;
+uint32_t           PageGuardManager::signal_handler_watcher_restores = 0;
+
+void PageGuardManager::MarkAllTrackedMemoryAsDirty()
+{
+    std::lock_guard<std::mutex> lock(tracked_memory_lock_);
+
+    for (auto entry = memory_info_.begin(); entry != memory_info_.end(); ++entry)
```

Comment:
Could this be a range-for?  (e.g. `for (auto& entry : memory_info_) { ... }` and then you'd change `entry->second` to `entry.second` )